images/spark/Dockerfile

```diff
@@ -0,0 +1,52 @@
+ARG SPARK_HOME=/opt/spark
+
+FROM alpine:3.10.2 AS spark-dist
+
+ARG SPARK_DIST_URL="http://archive.apache.org/dist/spark/spark-2.4.4/spark-2.4.4-bin-hadoop2.7.tgz"
+ARG SPARK_HOME
```

Comment:
looks like a duplicate `ARG`

Reply:
yes, it looks like it is, but this is actually the way to use global `ARG` in Docker: https://docs.docker.com/engine/reference/builder/#understand-how-arg-and-from-interact

Reply:
got it, thanks. Although yet another question here: why do we need multi-stage build here? From what I can tell we just download and unpack a tarball, what are the benefits? Caching?

Reply:
Yes, in that particular case multi-stage build doesn't introduce any significant benefits, but from my perspective it has some pros:
- the final image is more lightweight, due to exclusion of the tools we need to download and verify Spark dist;
- we don't need to do manual cleanup - just pick what we need we need and discard all the rest.
But this approach has one big drawback - it leaves intermediate image, which needs to be cleaned manually. Therefore clean-up is needed one way or another. Given that, I'm considering to refactor the `Dockerfile` and perform build as single stage.